Documentation/Evolution/CharacterClasses.md

```diff
@@ -0,0 +1,481 @@
+  /// characters, are considered vertical whitespace:
+  ///
+  /// - `LINE FEED (LF)` (U+000A)
+  /// - `LINE TABULATION` (U+000B)
+  /// - `FORM FEED (FF)` (U+000C)
+  /// - `CARRIAGE RETURN (CR)` (U+000D)
+  /// - `NEWLINE (NEL)` (U+0085)
+  public var isVerticalWhitespace: Bool { get }
+}
+```
+
+`\S`, `\H`, and `\V` match the inverse of `\s`, `\h`, and `\v`, respectively.
+
+We propose `\R` include "verticalWhitespace" above with detection (and consumption) of the CR-LF sequence when applied to `Unicode.Scalar`. It is equivalent to `Character.isVerticalWhitespace` when applied to `Character`s.
+
+We are similarly not proposing any new API for `\R` until the stdlib has grapheme-breaking API over `Unicode.Scalar`.
+
+_<details><summary>Rationale</summary>_
+
+We chose the Unicode recommendation as the basis for `Unicode.Scalar`, which is to derive whitespace matching from the Unicode `White_Space` property and general categories. This behavior matches `NSRegularExpression` and many languages with regular expressions in Unicode mode. In some languages, such as Go, `\s` is interpreted only as the ASCII whitespace characters, which would be surprising for Swift users. For details, see the [Unicode property list][proplist].
+
+We chose to treat any grapheme cluster that leads with whitespace to be a `Character`, for compatibility with existing `Character` API.
+
+</details>
+
+### Control characters: `\t`, `\r`, `\n`, `\f`, `\0`, `\e`, `\a`, `\b`, `\cX`
+
+We propose the following names and meanings for these escaped literals representing specific control characters:
+
+```swift
+extension Character {
+  /// TODO  
+  public static var tab: Character { get }
+
+  /// TODO   
+  public static var carriageReturn: Character { get }
+
+  /// TODO   
+  public static var lineFeed: Character { get }
+
+  /// TODO   
+  public static var formFeed: Character { get }
+
+  /// TODO   
+  public static var nul: Character { get }
+
+  /// TODO   
+  public static var escape: Character { get }
+
+  /// TODO   
+  public static var bell: Character { get }
+
+  /// TODO   
+  public static var backspace: Character { get }
+
+  /// TODO
+  public static func control(_ x: Unicode.Scalar): Character { }
+}
+
+extension Unicode.Scalar {
+  /// TODO  
+  public static var tab: Unicode.Scalar { get }
+
+  /// TODO   
+  public static var carriageReturn: Unicode.Scalar { get }
+
+  /// TODO   
+  public static var lineFeed: Unicode.Scalar { get }
+
+  /// TODO   
+  public static var formFeed: Unicode.Scalar { get }
+
+  /// TODO   
+  public static var nul: Unicode.Scalar { get }
+
+  /// TODO   
+  public static var escape: Unicode.Scalar { get }
+
+  /// TODO   
+  public static var bell: Unicode.Scalar { get }
+
+  /// TODO   
+  public static var backspace: Unicode.Scalar { get }
+
+  /// TODO
+  public static func control(_ x: Unicode.Scalar): Unicode.Scalar { }
+}
+```
+
+**TODO**: What about `\r\n` in grapheme semantic mode?
```

Comment:
**TODO**

Reply:
We probably want a static var for crlf and it would probably be in that list on Characters.